File: Problem-014/solution.py
def longest_collatz_sequence(limit):
    longest = 0
    answer = 0

    for n in range(1, limit):
        num = n
        length = 1

        while num != 1:
            if num % 2 == 0:
                num = num // 2
            else:
                num = 3 * num + 1

            length += 1

        if length > longest:
            longest = length
            answer = n

    return answer

File: Problem-014/solution.js
function longestCollatzSequence(limit) {
    let longest = 0;
    let answer = 0;

    for (let n = 1; n < limit; n++) {
        let num = n;
        let length = 1;

        while (num !== 1) {
            if (num % 2 === 0) {
                num = num / 2;
            } else {
                num = 3 * num + 1;
            }

            length++;
        }

        if (length > longest) {
            longest = length;
            answer = n;
        }
    }

    return answer;
}

